extensions/policies/src/main/java/org/eclipse/dataspaceconnector/mvd/RegionConstraintFunction.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.mvd;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.identityhub.credentials.model.VerifiableCredential;
+import org.eclipse.dataspaceconnector.policy.model.Operator;
+import org.eclipse.dataspaceconnector.policy.model.Permission;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.policy.AtomicConstraintFunction;
+import org.eclipse.dataspaceconnector.spi.policy.PolicyContext;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class RegionConstraintFunction implements AtomicConstraintFunction<Permission> {
+    private static final String VERIFIABLE_CREDENTIAL_KEY = "vc";
+    private static final String REGION_KEY = "region";
+    private final ObjectMapper objectMapper;
+    private final Monitor monitor;
+
+    public RegionConstraintFunction(ObjectMapper objectMapper, Monitor monitor) {
+        this.objectMapper = objectMapper;
+        this.monitor = monitor;
+    }
+
+    @Override
+    public boolean evaluate(Operator operator, Object rightValue, Permission rule, PolicyContext context) {
+        var regions = getRegions(context.getParticipantAgent().getClaims());
+        switch (operator) {
+            case EQ:
+                return regions.contains(rightValue);
+            default:
+                return false;
+        }
+    }
+
+    List<String> getRegions(Map<String, Object> claims) {
+        return claims.values()
+                .stream().flatMap(x -> getVerifiableCredential(x).stream())
```

Comment:
nit:
```
claims.values()
                .stream()
                .map(o -> (Map<String, Object>)o)
                .flatMap(x -> getVerifiableCredential(x).stream())
                .flatMap(vc -> getRegion(vc).stream())
                .collect(Collectors.toList());
```

this allows then to have `getVerifiableCredential(Map<String, Object> vcObject)` in the method below

Reply:
I prefer to keep the casting in a method to catch the exception at the right moment of the flow.